src/DGtal/geometry/surfaces/estimation/PlaneProbingDigitalSurfaceLocalEstimator.h

```diff
@@ -0,0 +1,402 @@
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file
+ * @author Jocelyn Meyron (\c jocelyn.meyron@liris.cnrs.fr )
+ * Laboratoire d'InfoRmatique en Image et Systemes d'information - LIRIS (CNRS, UMR 5205), CNRS, France
+ *
+ * @date 2020/12/07
+ *
+ * Header file for module PlaneProbingDigitalSurfaceLocalEstimator.cpp
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(PlaneProbingDigitalSurfaceLocalEstimator_RECURSES)
+#error Recursive header files inclusion detected in PlaneProbingDigitalSurfaceLocalEstimator.h
+#else // defined(PlaneProbingDigitalSurfaceLocalEstimator_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define PlaneProbingDigitalSurfaceLocalEstimator_RECURSES
+
+#if !defined PlaneProbingDigitalSurfaceLocalEstimator_h
+/** Prevents repeated inclusion of headers. */
+#define PlaneProbingDigitalSurfaceLocalEstimator_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include <unordered_map>
+#include "DGtal/base/Common.h"
+#include "DGtal/geometry/surfaces/DigitalSurfacePredicate.h"
+#include "DGtal/geometry/surfaces/estimation/MaximalSegmentSliceEstimation.h"
+#include "DGtal/topology/KhalimskyCellHashFunctions.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  // template class PlaneProbingDigitalSurfaceLocalEstimator
+  /**
+   * Description of template class 'PlaneProbingDigitalSurfaceLocalEstimator' <p>
+   * \brief Aim: Adapt a plane-probing estimator on a digital surface to estimate normal vectors.
+   *
+   * This class uses a plane-probing algorithm (whose type is given by the template parameter TProbingAlgorithm) to estimate
+   * normal vectors on a digital surface per surfel.
+   *
+   * @tparam TSurface the digital surface type.
+   * @tparam TProbingAlgorithm the probing algorithm (see \ref PlaneProbingTetrahedronEstimator or PlaneProbingParallelepipedEstimator).
+   *
+   * \b Models: A PlaneProbingDigitalSurfaceLocalEstimator is a model of concepts::CSurfelLocalEstimator and concepts::CDigitalSurfaceLocalEstimator.
+   */
+  template <typename TSurface, typename TProbingAlgorithm>
+  class PlaneProbingDigitalSurfaceLocalEstimator
+  {
+    // ----------------------- Public types ------------------------------
```

Comment:
I added the concept check for `TSurface::DigitalSurfaceContainer`.

It could be useful to have a concept for plane-probing algorithms, I'll have a look.